### PR TITLE
Feat: support cloud_id / cloud_auth configuration

### PR DESCRIPTION
**Pull request by kares**
heavily inspired by the ES output plugin: https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/906

docs all the same

### CODE REVIEW

**Comment by robbavey** [comment]
@karenzone Can you double check the docs here?

File: docs/index.asciidoc [hunk 1]
```diff
@@ -299,6 +301,26 @@ Tags the event on failure to look up previous log event information. This can be
...
 
+[id="plugins-{type}s-{plugin}-cloud_auth"]
+===== `cloud_auth`
+
+  * Value type is <<password,password>>
+  * There is no default value for this setting.
+
+Cloud authentication string ("<username>:<password>" format) is an alternative for the `user`/`password` pair.
+
+For mode details, check out the https://www.elastic.co/guide/en/logstash/current/connecting-to-cloud.html#_cloud_auth[cloud documentation]
```
Comment (karenzone) [inline docs/index.asciidoc]:
```suggestion
For mode details, check out the https://www.elastic.co/guide/en/logstash/current/connecting-to-cloud.html#_cloud_auth[Logstash-to-Cloud documentation]
```

File: docs/index.asciidoc [hunk 2]
```diff
@@ -299,6 +301,26 @@ Tags the event on failure to look up previous log event information. This can be
...
+
+[id="plugins-{type}s-{plugin}-cloud_id"]
+===== `cloud_id`
+
+  * Value type is <<string,string>>
+  * There is no default value for this setting.
+
+Cloud ID, from the Elastic Cloud web console. If set `hosts` should not be used.
+
+For mode details, check out the https://www.elastic.co/guide/en/logstash/current/connecting-to-cloud.html#_cloud_id[cloud documentation]
```
Comment (karenzone) [inline docs/index.asciidoc]:
```suggestion
For mode details, check out the https://www.elastic.co/guide/en/logstash/current/connecting-to-cloud.html#_cloud_id[Logstash-to-Cloud documentation]
```

**Comment by kares** [comment]
great, thanks for the reviews :bowing_man:

**Pull request merged**